control-plane/catalog/to-consul/resource_test.go

```diff
@@ -2154,11 +2154,22 @@ func createEndpointSlice(t *testing.T, client *fake.Clientset, serviceName strin
 					Conditions: discoveryv1.EndpointConditions{
 						Ready:       ptr.To(true),
 						Serving:     ptr.To(true),
-						Terminating: ptr.To(false),
+						Terminating: nil,
 					},
 					NodeName: &node2,
 					Zone:     ptr.To("us-west-2b"),
 				},
+				{
+					Addresses: []string{"3.3.3.3"},
+					Conditions: discoveryv1.EndpointConditions{
+						Ready:       ptr.To(false),
+						Serving:     ptr.To(false),
+						Terminating: ptr.To(true),
+					},
+					TargetRef: &targetRef,
+					NodeName:  &node1,
+					Zone:      ptr.To("us-west-2a"),
+				},
```

Comment:
With the updates to this function I think we'd need to update assertions. Can you run the tests in this file locally? I think a bunch will fail.

For example `TestServiceResource_nodePort` uses `createEndpointSlice`, but it checks that the list of endpoints registered has length 2 and doesn't have any assertions on the health of those endpoints. I think to all the functions that use `createEndpointSlice`, we need to update the length check to 3 and add assertions for the 3rd endpoint, and add health assertions for all 3 endpoints. 

```go
	actual := syncer.Registrations
	require.Len(r, actual, 3) // this should be 3 instead of 2
	require.Equal(r, "foo", actual[0].Service.Service)
	require.Equal(r, "1.2.3.4", actual[0].Service.Address)
	require.Equal(r, 30000, actual[0].Service.Port)
	require.Equal(r, "k8s-sync", actual[0].Node)
	require.Equal(r, "foo", actual[1].Service.Service)
	require.Equal(r, "2.3.4.5", actual[1].Service.Address)
	require.Equal(r, 30000, actual[1].Service.Port)
	require.Equal(r, "k8s-sync", actual[1].Node)
	// these should get added 
	require.Equal(r, "foo", actual[2].Service.Service)
	require.Equal(r, "1.2.3.4", actual[2].Service.Address) // since it's on node 1
	require.Equal(r, 30000, actual[2].Service.Port)
	require.Equal(r, "k8s-sync", actual[2].Node)
        // we should also assert on the health status being registered to consul
```

Reply:
Updated these and everything passes now